src/server/game/Entities/Player/Player.h

```diff
@@ -1093,6 +1093,8 @@ class TC_GAME_API Player : public Unit, public GridObject<Player>
         void SetGMVisible(bool on);
         void SetPvPDeath(bool on) { if (on) m_ExtraFlags |= PLAYER_EXTRA_PVP_DEATH; else m_ExtraFlags &= ~PLAYER_EXTRA_PVP_DEATH; }
 
+        uint32 GetXP() const { return m_activePlayerData->XP; }
```

Comment:
btw 335 has `return GetUInt32Value(PLAYER_XP)`

Reply:
Wait, so it is `GetUInt32Value`? Man I never understand legacy UF impl. xD